### PR TITLE
fixing broken links

### LINKED ISSUE #4013: Broken link on Kubernetes services page
I noticed a broken link on one of the docs pages. It's the link to code samples in the "What's Next" section on the Kubernetes services page: https://knative.dev/docs/serving/knative-kubernetes-services/

When clicked, the link gives a 404. It appears that the link is incorrect (it goes to https://knative.dev/docs/serving/knative-kubernetes-services/samples/ but really should be https://knative.dev/docs/serving/samples/, i.e. without the `knative-kubernetes-services` part).

Though when I checked the source, the link actually redirects back to the `samples/` directory, which I assume is what is supposed to happen. Don't have a build environment for the docs set up, so wasn't able to dig into it any further...

### PR DESCRIPTION
**Pull request by psschwei**
Fixes #4013 

## Proposed Changes 

- Fixes broken links on docs/serving/knative-kubernetes-services.md page
(note that the first link here was circular... I assume it was meant to point to the readme)
- Fixes broken links in `DEVELOPMENT.md`

### CODE REVIEW

**Review by abrennan89 — APPROVED** [review]
/lgtm

**Comment by abrennan89** [comment]
@psschwei this looks like it needs a rebase, also, does this need to be cherrypicked to 0.24 as well?

**Comment by psschwei** [comment]
Yes, this should be cherrypicked to 0.24, as the links are broken there as well

Will rebase shortly.

**Comment by psschwei** [comment]
@abrennan89 I think the last two links (to website / maintainer help) on `DEVELOPMENT.md` have been sunset, maybe as part of #4020 or #4024 .  I wonder if it would be better for it to mirror the changes @RichardJJG made to `blog/README.md`?

**Comment by abrennan89** [comment]
Assigning @RichardJJG to take a look - might be better to rebase once any of his PRs are merged?

**Comment by psschwei** [comment]
> @psschwei if this is undoing any of Richard's changes I'd just re-add whatever he did.

Not undoing any changes, at least I don't think it is. I think basically what's happening is:

- previously, the [`DEVELOPMENT.md`](https://github.com/knative/docs/blob/mkdocs/DEVELOPMENT.md) doc and the [`blog/README.md`](https://github.com/knative/docs/blob/21fe8c783ac0024d7b6f07cc3fc7f0e60d43255e/blog/README.md) doc were exactly the same
- in #4024, the `blog/README.md` doc was truncated to a single, main link, but `DEVELOPMENT.md` was not changed
- my question was whether the [`DEVELOPMENT.md`](https://github.com/knative/docs/blob/mkdocs/DEVELOPMENT.md) doc should also be truncated to the single main link to the "contributing to knative docs" documentation, like the [new `blog/README.md`](https://github.com/knative/docs/blob/mkdocs/blog/README.md)?

**Comment by abrennan89** [comment]
cc @omerbensaadon @RichardJJG to answer the comments / questions on this one please

**Comment by omerbensaadon** [comment]
@psschwei it might be easier to revert changes to the DEVELOPMENT.md file and just ship the fix the linked issue is targeting

File: DEVELOPMENT.md
Comment (omerbensaadon) [inline DEVELOPMENT.md:22]: These pages don't exist anymore
```suggestion
```

**Review by omerbensaadon — CHANGES_REQUESTED** [review]
I think if you remove this it should be good to merge? If you don't have any conflicts with Richard's changes, I'd /unhold. 

If you do, I'd take @RichardJJG's changes where there are conflicts.

**Comment by psschwei** [comment]
It looks like #4054 also fixes the original broken links (and in a better way than my PR), so happy to close this one and let the other one merge instead.

(And apologies to @snneji  for forgetting to assign the issue)

**Comment by omerbensaadon** [comment]
I think your PR still does fix some broken links in Development.MD as well as the blog README

**Comment by omerbensaadon** [comment]
If you can 

1. Remove the last two lines from my review 
2. Remove changes covered by #4054 

We can get this merged

**Comment by psschwei** [comment]
> /hold

Looks like both @abrennan89 and I put holds on this...  I'm good to remove mine.

**Review by abrennan89 — APPROVED** [review]
/lgtm
/unhold

**Pull request merged**
